src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2113,8 +2113,22 @@ def sync_db_keys(self, job, name=None):
             # We don't want to do this for passive controller
             return
         filters = [['OR', [['name', '=', name], ['name', '^', f'{name}/']]]] if name else []
+
+        # It is possible we have a pool configured but for some mistake/reason the pool did not import like
+        # during repair disks were not plugged in and system was booted, in such cases we would like to not
+        # remove the encryption keys from the database.
+        for root_ds in {pool['name'] for pool in self.middleware.call_sync('pool.query')} - {
+            ds['id'] for ds in self.middleware.call_sync(
+                'pool.dataset.query', [], {'extra': {'retrieve_children': False, 'properties': []}}
+            )
+        }:
+            filters.extend([['name', '!=', root_ds], ['name', '!=', f'{root_ds}/']])
```

Comment:
Easier/better to have a single filter with `['name', 'nin', list_after_all_this_loop]`' ?

Reply:
That would mean that we should have the list of encrypted datasets in that specific pool which did not import at the time ?

Reply:
I am not suggesting any logic change. Isn;t what I proposed exactly the same but using a single filter instead of multiple?